wp/lib/bap_wp/src/constraint.ml

```diff
@@ -246,12 +246,12 @@ let rec eval_aux ?stats:(stats = init_stats) (constr : t) (olds : z3_expr list)
       (eval_aux ~stats:stats c2 olds news ctx)
   | Clause (hyps, concs) ->
     let eval_conjunction conj =
-      if List.length conj = 1 then 
-        (* This is tail recursive. Avoids And node. *)
-        eval_aux ~stats:stats (List.hd_exn conj) olds news ctx 
-      else
+      (* This adds possible tail recursiion, avoids And node. *)
```

Comment:
"adds possible"? Is it tail recursive or not?

Also: `recursiion`.

Reply:
Some branches are tail recursive, some aren't. Depends on the input